src/report.rs

```diff
@@ -0,0 +1,172 @@
+// SPDX-License-Identifier: Apache-2.0
+
+//! Intel SGX Enclave report structures.
+
+use crate::parameters::{Attributes, Features, MiscSelect, Xfrm};
+
+use core::fmt::{self, Display};
+
+#[derive(Debug, PartialEq)]
+#[non_exhaustive]
+pub enum ReportError {
+    InvalidMiscSelect(u32),
+    InvalidFeatures(u64),
+    InvalidXfrm(u64),
+}
+
+impl Display for ReportError {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self {
+            ReportError::InvalidMiscSelect(value) => {
+                write!(f, "Invalid misc select: {}", value)
+            }
+            ReportError::InvalidFeatures(value) => {
+                write!(f, "Invalid features: {}", value)
+            }
+            ReportError::InvalidXfrm(value) => {
+                write!(f, "Invalid xfrm: {}", value)
+            }
+        }
+    }
+}
+
+/// The enclave report body.
+///
+/// For more information see the following documents:
+///
+/// [Intel® Software Guard Extensions (Intel® SGX) Data Center Attestation Primitives: ECDSA Quote Library API](https://download.01.org/intel-sgx/dcap-1.0/docs/SGX_ECDSA_QuoteGenReference_DCAP_API_Linux_1.0.pdf)
+///
+/// Table 5, A.4. Quote Format
+///
+/// [Intel® 64 and IA-32 Architectures Software Developer's Manual Volume 3 (3A, 3B, 3C & 3D): System Programming Guide](https://www.intel.com/content/www/us/en/architecture-and-technology/64-ia-32-architectures-software-developer-vol-3d-part-4-manual.html)
+///
+/// Table 38-21. Layout of REPORT
+#[derive(Clone, Debug)]
+#[repr(C)]
+pub struct ReportBody {
+    cpusvn: [u8; 16],
+    miscselect: u32,
+    reserved1: [u8; 28],
+    // The attributes field is composed of the features and xfrm fields.
+    features: u64,
```

Comment:
I think this should be `[u8; 8]`
We will also need to add the `align` attribute to the struct if we want to preserve the alignment

Reply:
I don't think we care how reportbody is aligned but I might be wrong. I'll ignore the alignment and we can change it later. I don't see anything in the docs about this specifically needing it.